### DISCUSSION TITLE
About the scope of the roadmap

### DISCUSSION THREAD

**Discussion by KaruroChori** [rysana-ai/bundown#6] [General]
I have few comments about the roadmap, and since discussions are not enabled in this repo, I will write them down here.

We are using markdown tags for bash and other shell variants to mark sections to be executed by the new bun shell syntax. 

However, those are neither bash nor zsh.  
For example, things like `for` cycles do not work, and the syntax is somewhat divergent from any specific shell, favoring the integration with js/ts. This design decision/limitation is totally fine within the scope of bun shell, but considering that you envision support for multiple languages as part of the roadmap, this would leave shell scripting far behind in terms of compatibility.

In my opinion, the only way to have a **true** portable bash would be to compile it for webassembly, and pack it with the application, or to manually exec it hoping the system has minimal support for it, which was one of the main reasons why bun shell was made in the first place.

**Comment by jrysana** [comment]
Thanks for opening these! Addressing a few of your points:

Regarding Discussions - thanks for noting, I didn't even notice they were off by default. Enabled now :)

## Bun Shell v.s. other shell and programming languages

To your points, I really like the fact that Bun Shell integrates nicely with the JS/TS core of Bun and is quite portable and compatible with all systems (soon even Windows), but it does have its tradeoffs.

It might just be my personal bias speaking, but for a number of reasons I think a good core for Bundown to maintain is a focus on the synergy between TS, JS, and Shell scripts that we can deliver over time. I don't however want to arbitrarily restrict people from being able to run whatever they like - be it Python, C/C++/Rust/Zig, etc., and ideally it would even be cool to have whatever level of automatic support for FFI bindings and cross-language variable sharing through some type of lightweight opt-out JSON serialization + macros approach or something like that.

We need to find the right balance, and I think both the ideas of packing a compiled version of a specific shell or invoking actual shells like Bash and Zsh directly through the local device have valid benefits but I'm unsure of where we should go on the shell languages front as of yet.

Having that generic cross-language binding layer for basic variables might tip things slightly in favor of supporting or defaulting to non Bun Shell execution for Bash/Zsh/etc, as well as having things like the automatic OS switch tagging listed in the roadmap.

Curious to hear more of your thoughts on this @KaruroChori
